source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -189,7 +189,8 @@ Http::FilterDataStatus StreamHandleWrapper::onData(Buffer::Instance& data, bool
     coroutine_.resume(0, yield_callback_);
   }
 
-  if (state_ == State::HttpCall || state_ == State::WaitForBody) {
+  if (state_ == State::HttpCall || state_ == State::WaitForBody ||
+      state_ == State::WaitForBodyChunk) {
```

Comment:
I might be missing some context, but I don't think this is a change that we want? The idea behind body chunks is to allow viewing the data without buffering. If the user wants the data buffered they should use body()? I think the other issue is perhaps not a bug and by design? wdyt?

Reply:
Looking at the issue once again and the docs. I agree, seems like this change is unnecessary. Will close it then.